spirv_common.hpp

```diff
@@ -296,6 +296,8 @@ struct SPIRType : IVariant
 		bool depth;
 		bool arrayed;
 		bool ms;
+		bool is_read = false;
```

Comment:
No...I don't want to make all textures writable.

Marking a texture as writable will in most cases cause Metal to optimize it differently, likely negatively relative to a read-only texture.

The MSL produced by CompilerMSL is used in production shaders, where performance is a concern. Writable textures are very rare, and making all textures writable will likely introduce a performance overhead on all basic sampled textures. I would like Metal to be free to optimize for those vast majority read-only cases.

For now...can we leave the mechanism as it is in this PR...and then move quickly to deploy a refactoring of `type_to_glsl()`?